addons/google_calendar/models/google_calendar.py

```diff
@@ -601,153 +593,146 @@ def synchronize_events(self, cr, uid, ids, lastSync=True, context=None):
-                if not source_attendee_record_id:
-                    continue
-                source_attendee_record = att_obj.browse(cr, uid, source_attendee_record_id, context)[0]
-
-                if att.event_id.recurrent_id_date and source_event_record.allday and source_attendee_record.google_internal_event_id:
-                    new_google_internal_event_id = source_attendee_record.google_internal_event_id + '_' + att.event_id.recurrent_id_date.split(' ')[0].replace('-', '')
-                elif att.event_id.recurrent_id_date and source_attendee_record.google_internal_event_id:
-                    new_google_internal_event_id = source_attendee_record.google_internal_event_id + '_' + att.event_id.recurrent_id_date.replace('-', '').replace(' ', 'T').replace(':', '') + 'Z'
-
-                if new_google_internal_event_id:
-                    #TODO WARNING, NEED TO CHECK THAT EVENT and ALL instance NOT DELETE IN GMAIL BEFORE !
-                    try:
-                        st, response, ask_time = self.update_recurrent_event_exclu(cr, uid, new_google_internal_event_id, source_attendee_record.google_internal_event_id, att.event_id, context=context)
-                        if status_response(st):
-                            att_obj.write(cr, uid, [att.id], {'google_internal_event_id': new_google_internal_event_id}, context=context)
-                            new_ids.append(new_google_internal_event_id)
-                            cr.commit()
-                        else:
-                            _logger.warning("Impossible to create event %s. [%s]" % (att.event_id.id, st))
-                            _logger.debug("Response : %s" % response)
-                    except:
-                        pass
-        return new_ids
+        CalendarEvent = self.env['calendar.event']
+        CalendarAttendee = self.env['calendar.attendee']
+        ResUsers = self.env['res.users']
+        my_partner_id = self.env.user.partner_id.id
 
-    def update_events(self, cr, uid, lastSync=False, context=None):
-        context = dict(context or {})
+        context_norecurrent = self.get_context_no_virtual()
+        my_attendees = CalendarAttendee.with_context(context_norecurrent).search([('partner_id', '=', my_partner_id), 
+                                    ('google_internal_event_id', '=', False),
+                                    ('event_id.recurrent_id', '!=', False),
+                                    ('event_id.recurrent_id', '>', 0)])
+        for att in my_attendees:
+            new_google_internal_event_id = False
+            source_event_record = CalendarEvent.browse(att.event_id.recurrent_id)
+            source_attendee_records = CalendarAttendee.search([('partner_id', '=', my_partner_id), ('event_id', '=', source_event_record.id)])
+            if not source_attendee_records:
+                continue
+            source_attendee_record = source_attendee_records[0]
```

Comment:
if you want perform operation on only first record on recordset then pass limit=1 on search and assign dircetly it to source_attendee_record